### PR TITLE
feat: Create CI scripts to test new nightly bundles using new Openshift CI bundles workflow

### PR DESCRIPTION
**Pull request by flacatus**
Reff issue: https://github.com/eclipse/che/issues/18293

Signed-off-by: Flavius Lacatusu <flacatus@redhat.com>

### CODE REVIEW

**Comment by tolusha** [comment]
Some general remarks:
- `.ci/common.sh` isn't needed, we have `.github/bin/common.sh`
- pls refactor `.ci/oci-single-host.sh` and `.ci/oci-nightly-olm.sh`  like `.ci/cico_openshift_shost.sh`:

**Comment by tolusha** [comment]
`olm/testCatalogSource.sh` script is mentioned in olm/README.md file
Will it work after the changes ?

File: .github/bin/minikube/provision-cluster.sh [hunk 1]
```diff
@@ -9,6 +9,7 @@
 #
 # Contributors:
 #   Red Hat, Inc. - initial API and implementation
+set -ex
 
 # Minikube environments config
 export MINIKUBE_VERSION=v1.8.2
```
Comment (tolusha) [inline .github/bin/minikube/provision-cluster.sh:15]: Is it too old ?

File: olm/eclipse-che-preview-kubernetes/Dockerfile [hunk 2]
```diff
@@ -1,17 +1,25 @@
...
+# which is available at https://www.eclipse.org/legal/epl-2.0/
+#
+# SPDX-License-Identifier: EPL-2.0
+#
+# Contributors:
+#   Red Hat, Inc. - initial API and implementation
 
-COPY deploy manifests/eclipse-che-preview-kubernetes
-RUN ./bin/initializer -o ./bundles.db
+FROM quay.io/operator-framework/upstream-registry-builder as builder
```
Comment (tolusha) [inline olm/eclipse-che-preview-kubernetes/Dockerfile:13]: Tag is forgotten

File: olm/eclipse-che-preview-openshift/Dockerfile [hunk 3]
```diff
@@ -1,17 +1,26 @@
...
+# This program and the accompanying materials are made
+# available under the terms of the Eclipse Public License 2.0
+# which is available at https://www.eclipse.org/legal/epl-2.0/
+#
+# SPDX-License-Identifier: EPL-2.0
+#
+# Contributors:
+#   Red Hat, Inc. - initial API and implementation
+
+FROM quay.io/operator-framework/upstream-registry-builder as builder
```
Comment (tolusha) [inline olm/eclipse-che-preview-openshift/Dockerfile:14]: Tag is forgotten

**Comment by flacatus** [comment]
> `olm/testCatalogSource.sh` script is mentioned in olm/README.md file
> Will it work after the changes ?

Should work

File: .github/bin/minikube/provision-cluster.sh (hunk 1 above)
Comment (flacatus) [reply .github/bin/minikube/provision-cluster.sh:15]: I'm changing the minikube version in stable tests.

Comment (flacatus) [reply .github/bin/minikube/provision-cluster.sh:15]: It will be a separate PR

File: .ci/oci-nightly-olm.sh [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+set -e
+# only exit with zero if all commands of the pipeline exit successfully
+set -o pipefail
+# error on unset variables
+set -u
+
+export OPERATOR_REPO=$(dirname $(dirname $(readlink -f "$0")));
+source "${OPERATOR_REPO}"/.github/bin/common.sh
+
+#Stop execution on any error
```
Comment (AndrienkoAleksandr) [inline .ci/oci-nightly-olm.sh:26]: Minor: #Stop -> # Stop

**Review by AndrienkoAleksandr — APPROVED** [review]
Impressive job!

File: olm/eclipse-che-preview-kubernetes/Dockerfile [hunk 5]
```diff
@@ -1,17 +1,25 @@
...
+# SPDX-License-Identifier: EPL-2.0
+#
+# Contributors:
+#   Red Hat, Inc. - initial API and implementation
 
-COPY deploy manifests/eclipse-che-preview-kubernetes
-RUN ./bin/initializer -o ./bundles.db
+FROM quay.io/operator-framework/upstream-registry-builder:v1.15.3 as builder
 
-FROM openshift/origin-base
```
Comment (tolusha) [inline olm/eclipse-che-preview-kubernetes/Dockerfile:6]: I am a bit concerned of changing this image.
Do we have any issues with the current one?

Comment (flacatus) [reply olm/eclipse-che-preview-kubernetes/Dockerfile:6]: The most examples in appr-registry are using https://github.com/operator-framework/operator-registry/blob/master/appr-registry.Dockerfile#L20

Comment (tolusha) [reply olm/eclipse-che-preview-kubernetes/Dockerfile:6]: Thank you. It make sense then.

**Review by tolusha — APPROVED** [review]

**Pull request merged**